src/firekit.js

```diff
@@ -0,0 +1,132 @@
+
+// Define the App Check debug token
+// Debug tokens are only allowed in the local development environment and in Cypress tests. If the token environment
+// variable is set to 'true', we pass on the boolean value in order to generate a new debug token. Otherwise, we pass on
+// the value of the environment variable to allow engineers to set their own personal debug token.
+const APP_CHECK_DEBUG_TOKEN =
+  isLocalhost || isCypress
+    ? VITE_FIREKIT_APPCHECK_DEBUG_TOKEN === 'true'
+      ? true
+      : VITE_FIREKIT_APPCHECK_DEBUG_TOKEN
+    : undefined;
+
+// Define the Firekit configuration object
+// If the project is running in the emulator, only a subset of the configuration is required. Otherwise, the ROAR
+// Firekit requires the full configuration object to be passed in during initialization.
+const firekitConfig = VITE_FIREBASE_EMULATOR_ENABLED
+  ? {
+      admin: {
+        projectId: `demo-${VITE_FIREBASE_ADMIN_PROJECT_ID}`,
+        apiKey: 'fake-admin-emulator-api-key',
+        emulatorPorts: {
+          db: AdminFirebaseConfig.emulators.firestore.port,
+          auth: AdminFirebaseConfig.emulators.auth.port,
+          functions: AdminFirebaseConfig.emulators.functions.port,
+        },
+      },
+      app: {
+        projectId: `demo-${VITE_FIREBASE_APP_PROJECT_ID}`,
+        apiKey: 'fake-app-emulator-api-key',
+        emulatorPorts: {
+          db: AppFirebaseConfig.emulators.firestore.port,
+          auth: AppFirebaseConfig.emulators.auth.port,
+          functions: AppFirebaseConfig.emulators.functions.port,
+        },
+      },
+    }
+  : {
+      admin: {
+        projectId: VITE_FIREBASE_ADMIN_PROJECT_ID,
+        appId: VITE_FIREBASE_ADMIN_APP_ID,
+        apiKey: VITE_FIREBASE_ADMIN_API_KEY,
+        authDomain: VITE_FIREBASE_ADMIN_AUTH_DOMAIN,
+        storageBucket: VITE_FIREBASE_ADMIN_STORAGE_BUCKET,
+        messagingSenderId: VITE_FIREBASE_ADMIN_MESSAGING_SENDER_ID,
+        siteKey: VITE_FIREKIT_ADMIN_RECAPTCHA_SITE_KEY,
+        debugToken: APP_CHECK_DEBUG_TOKEN,
+        measurementId: VITE_ANALYTICS_ADMIN_MEASUREMENT_ID,
+      },
+      app: {
+        projectId: VITE_FIREBASE_APP_PROJECT_ID,
+        appId: VITE_FIREBASE_APP_APP_ID,
+        apiKey: VITE_FIREBASE_APP_API_KEY,
+        authDomain: VITE_FIREBASE_APP_AUTH_DOMAIN,
+        storageBucket: VITE_FIREBASE_APP_STORAGE_BUCKET,
+        messagingSenderId: VITE_FIREBASE_APP_MESSAGING_SENDER_ID,
+        siteKey: VITE_FIREKIT_APP_RECAPTCHA_SITE_KEY,
+        debugToken: APP_CHECK_DEBUG_TOKEN,
+      },
+    };
+
+/**
+ * Initialize a new Firekit instance.
+ *
+ * @returns {Promise<RoarFirekit>} A promise that resolves with the initialized Firekit instance.
+ */
+export async function initializeFirekit() {
+  const firekit = new RoarFirekit({
+    roarConfig: firekitConfig,
+    authPersistence: 'session',
+    markRawConfig: {
+      // @TODO: Check what this is for?
+      auth: false,
+      db: false,
+      functions: false,
```

Comment:
There was a time when we though we had to use `markRaw` for all of the Firebase service class instances. So we build this into firekit:

- See [here](https://github.com/yeatmanlab/roar-firekit/blob/21292c4782f1f15696f94cb09539e5cc78f2c113/src/firekit.ts#L209)
- And [here](https://github.com/yeatmanlab/roar-firekit/blob/21292c4782f1f15696f94cb09539e5cc78f2c113/src/firestore/util.ts#L157)

But as you can see, we don't use this. I think we could remove this config here since it is the default behavior. Could you also submit a firekit PR removing this config entirely. I don't believe we use it anywhere and perhaps we shouldn't maintain it if we never intend to use it.

Reply:
Thanks for the explanation! I'll open separate PRs to remove this from the dashboard and firekit repos shortly